Dockerfile

```diff
@@ -0,0 +1,89 @@
+# syntax=docker/dockerfile:1.2
+
+# Full Build - This container will do a full compile and create a minimal runtime image.
+
+# This build requires BuildKit
+
+# To build testnet
+#   DOCKER_BUILDKIT=1 docker build -t mobilecoin/full-service:0.0.0-testnet \
+#   --progress=plain --build-arg NAMESPACE=test --build-arg BUILD_OPTS=--no-default-features .
+
+# To build mainnet
+#   DOCKER_BUILDKIT=1 docker build -t mobilecoin/full-service:0.0.0 \
+#   --progress=plain --build-arg NAMESPACE=prod .
+
+# Build Args:
+# BUILD_OPTS: - '--no-default-features' - Additonal options to cargo build command
+# NAMESPACE: - test|prod - specifies enclave.css files to download.
+# SGX_MODE: - HW|SW - See README
+# IAS_MODE: - PROD|DEV - See README
+# RUSTFLAGS: - '-C target-cpu=penryn' - RUSTFLAGS Environment Variable
+# SIGNED_ENCLAVE_BASE: - enclave-distribution.${NAMESPACE}.mobilecoin.com - base domain for CSS files.
+# SIGSTRUCT_JSON_LOCATION: - production.json - Json file where CSS file paths are located.
+
+# IMPORTANT: Do not add or update OS packages or components in the builder section.
+# In order to create a consistent and verifiable the build environment, only add
+# or update in the mobilecoin/rust-sgx-base image and refer to the image by its hash.
+
+FROM mobilecoin/rust-sgx-base@sha256:5ffeda68b4a3f28370bd2a533c8ada0f95cbc9b8232b8cd5a633b47a19e39cfc as builder
+
+ARG NAMESPACE=test
+ARG SIGNED_ENCLAVE_BASE=enclave-distribution.${NAMESPACE}.mobilecoin.com
+ARG SIGSTRUCT_JSON_LOCATION=production.json
+
+ENV INGEST_ENCLAVE_CSS=/app/ingest-enclave.css
+ENV CONSENSUS_ENCLAVE_CSS=/app/consensus-enclave.css
+
+WORKDIR /app
+
+# Get enclave sigstruct
+RUN  export SIGSTRUCT_JSON=$(curl -sSL https://${SIGNED_ENCLAVE_BASE}/${SIGSTRUCT_JSON_LOCATION}) \
+  && export CONSENSUS_CSS_URL=$(echo ${SIGSTRUCT_JSON} | jq -r .consensus.sigstruct) \
+  && export INGEST_CSS_URL=$(echo ${SIGSTRUCT_JSON} | jq -r .ingest.sigstruct) \
+  && curl https://${SIGNED_ENCLAVE_BASE}/${CONSENSUS_CSS_URL} -o ${CONSENSUS_ENCLAVE_CSS} \
+  && curl https://${SIGNED_ENCLAVE_BASE}/${INGEST_CSS_URL} -o ${INGEST_ENCLAVE_CSS}
+
+COPY . /app/
+
+ARG BUILD_OPTS
+ARG SGX_MODE=HW
+ARG IAS_MODE=PROD
+ARG RUSTFLAGS='-C target-cpu=penryn'
```

Comment:
We've been using skylake in other places. How was this chosen? Whats the purpose of this container?

Reply:
moved to skylake